tools/cli/commands/list.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for implementing the `datalab list` command."""
+
+import argparse
+
+
+_FILTER_HELP=(
+"""Apply a Boolean filter EXPRESSION to each resource item
+to be listed.
+
+If the expression evaluates True then that item is listed.
+For more details run `gcloud topic filters`."""
+)
+
+
+_ZONES_HELP=(
+"""List of zones to which to limit the resulting list."""
+)
+
+
+description=(
+"""{0} {1} displays the Datalab instances running in Google
+Compute Engine VM's in a project.
+
+By default, instances from all zones are listed. The results
+can be narrowed down by providing the --zones flag."""
+)
+
+
+examples=("""
+To list all of the available Datalab instances in a project:
+
+    $ {0} {1}
+
+To only list the Datalab instances in the zones
+'us-central1-a' and  'us-central1-b':
+
+    $ {0} {1} --zones us-central1-a,us-central1-b
```

Comment:
So I personally like my single fully qualified name pattern, and if you agree, I think this would just be either:

xyzproject.us-central1-a to list all in a zone or
xyzproject to list all in a project and configured default zone
or nothing to list all in configured project and zone

So just:

    datalab list
    datalab list xyzproject.us-central1-a

Thoughts?